### PR TITLE
Gazebo 9 compatibility (take 2)

### PR DESCRIPTION
**Pull request by julianoes**
Switch from Gazebo math to ignition math

This makes the plugins compatible with Gazebo 9.

These changes were already done by @william-peale before but never merged.
Instead of fixing the conflicts, I decided to re-apply the changes again cleanly with this commit.

Replaces #109.

Needs testing with Gazebo 7 and 8.

### CODE REVIEW

**Comment by julianoes** [comment]
Ok looks like the build is broken for Gazebo < 9. I'll look into it.

File: include/common.h [hunk 1]
```diff
@@ -52,7 +53,7 @@ bool getSdfParam(sdf::ElementPtr sdf, const std::string& name, T& param, const T
 /**
  * \brief Get a math::Angle as an angle from [0, 360)
  */
-double GetDegrees360(const math::Angle& angle) {
+double GetDegrees360(const ignition::math::Angle& angle) {
```
Comment (TSC21) [inline include/common.h:56]: @julianoes can we add `using namespace ignition::math`

**Review by TSC21 — CHANGES_REQUESTED** [review]

**Comment by TSC21** [comment]
We probably need to setup different CI instances for the different Gazebo versions. @dagar do you agree on setting up new containers for the different versions?

File: include/common.h (hunk 1 above)
Comment (julianoes) [reply include/common.h:56]: You should not use `using namespace` in header files.

Comment (TSC21) [reply include/common.h:56]: Yeah sorry you are right I didn't see that this was being changed on headers only.

**Review by TSC21 — APPROVED** [review]
Just needs some testing probably but structuraly looks good to me.

**Comment by julianoes** [comment]
Tested successfully with Gazebo 7.11.0 on macOS, Gazebo 9.0.0 on Arch Linux and Gazebo 8.1.1 on Fedora.

**Comment by TSC21** [comment]
Will do a test today on Ubuntu with Gazebo 7

**Comment by TSC21** [comment]
Will do a test today on Ubuntu with Gazebo 7

**Comment by TSC21** [comment]
@julianoes feel free to delete the dev branch you used. And thanks for the upgrade!

**Comment by julianoes** [comment]
@TSC21 thanks for reviewing and merging.

**Comment by hamishwillee** [comment]
@julianoes @TSC21 As you know, docs and install scripts say we support Gazebo 8 for "standalone" Gazebo. For ROS (Indigo) we support Gazebo 7, as that is the compatible version. 

How does this update change things? Are we still _supporting_ Gazebo 8 or is our supported set now Gazebo 9 standalone and Gazebo 7 for ROS. 

Is MacOS dev environment installer also updated to Gazebo 9?

**Comment by julianoes** [comment]
This is backwards compatible with Gazebo 7 and Gazebo 8 so we don't have to change anything (just yet). But some have these rolling release distributions that almost force you to always use the latest and greatest :D.

**Comment by hamishwillee** [comment]
@julianoes Thanks. Just so I can be precise - is MacOS dev environment installer also updated to Gazebo 9?

**Comment by hamishwillee** [comment]
I can't get Gazebo9 to build. Error in this GIST https://gist.github.com/hamishwillee/14c8a2d5adda9801e45f425dd80d7616 , setup as per https://github.com/PX4/Devguide/pull/497

**Comment by julianoes** [comment]
> @julianoes Thanks. Just so I can be precise - is MacOS dev environment installer also updated to Gazebo 9?

No not yet. I'm not trying to move everything to 9, I just had Gazebo 9 on my machine and made changes to make it work.

**Comment by julianoes** [comment]
> @hamishwillee I can't get Gazebo9 to build. Error in this GIST https://gist.github.com/hamishwillee/14c8a2d5adda9801e45f425dd80d7616 , setup as per PX4/Devguide#497

Are you sure you actually used the latest sitl_gazebo master?

**Comment by hamishwillee** [comment]
Good point - no I used Firmware master, and expected it to bring me the correct version of sitl_gazebo.  When is that likely to be true (ie when will fetching fresh Firmware get me this version of gazebo?)

**Comment by julianoes** [comment]
I forgot to do a PR, done now: https://github.com/PX4/Firmware/pull/9177
(We should automate this using e.g. https://renovateapp.com/)

**Comment by hamishwillee** [comment]
Thanks. It works now.

**Pull request merged**